### PR TITLE
Run XO as root and add NFS

### PR DESCRIPTION
**Pull request by MikeDombo**
Not sure if you, Ezka77 want this, but this PR enables NFS remotes.

Run XO as root and added NFS service to enable backups to NFS remotes.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -22,20 +22,20 @@ FROM node:6-alpine
...
+ENV USER=root \
+    USER_HOME=/root \
     XOA_PLAN=5 \
     DEBUG=xo:main
 
-WORKDIR /home/node
+WORKDIR /root
 
-RUN apk add --no-cache tini su-exec bash util-linux
+RUN apk add --no-cache tini su-exec bash util-linux nfs-utils openrc
```
Comment (Ezka77) [inline Dockerfile:32]: I see the add of `nfs-utils` (which I understand we need the nfs tools to use NFS) and `openrc`  ? why the  `openrc` package ?

File: Dockerfile [hunk 2]
```diff
@@ -52,13 +52,14 @@ RUN npm install --global \
     xo-server-load-balancer \
     xo-import-servers-csv
 
+RUN rc-update add nfsmount
```
Comment (Ezka77) [inline Dockerfile:55]: your enabling `nfsmount` on start ... well great but we never start any rc daemon in the container, the `ENTRYPOINT` use `tini` to start the `entrypoint.sh` script, and keep running this script. So `nfsmount` should not run on start.

File: xo-entry.sh [hunk 3]
```diff
@@ -1,5 +1,7 @@
 #!/bin/bash
 set -e
 
-exec su-exec ${USER} "$@"
+rpcbind -f &
```
Comment (Ezka77) [inline xo-entry.sh:4]: I think `rpcbind` is added by the `openrc` package, is that explain the openrc ? What seems odd to me is that `rpcbind` is a server, using `nfsmount` require `rpcbind` ?

File: Dockerfile [hunk 4]
```diff
@@ -22,20 +22,20 @@ FROM node:6-alpine
 LABEL xo-server=5.18.3 \
          xo-web=5.18.2
 
-ENV USER=node \
-    USER_HOME=/home/node \
+ENV USER=root \
```
Comment (Ezka77) [inline Dockerfile:25]: this configuration set the environment variable `USER` you can overwrite it on your `docker-compose.yml` or change this, like do in the build file, and the application should run with that user. So a simple `USER=root` should be enough to run the application with the root user.

File: Dockerfile [hunk 5]
```diff
@@ -1,15 +1,15 @@
 # Build container
 FROM node:6-alpine as build_container
 
-WORKDIR /home/node
+WORKDIR /root
```
Comment (Ezka77) [inline Dockerfile:4]: Keep all the build in `/home/node`: building and running is a different matter, it should not affect the way you start the application in the end and it avoid most of the fuzz of your PR.

File: Dockerfile [hunk 6]
```diff
@@ -1,15 +1,15 @@
...
 
 RUN apk add --no-cache git python g++ make bash libc6-compat
 
-USER node
+USER root
```
Comment (Ezka77) [inline Dockerfile:8]: Keep using `node` as user, in Dokerfile you can make all your build with users privilege.

**Review by Ezka77 — COMMENTED** [review]
Hi Mike,
I have to say I'm not a fan to start things in root, but enabling backups to remote NFS is a good idea =). Let me tear down your PR and maybe we can find a way to keep using the 'node' user and enabling remote NFS backups.

**Comment by MikeDombo** [comment]
Yeah, I know it isn't good practice, but I think it is necessary since xo
runs `mount` which needs root to work. But if you can find a better way,
then that's awesome.

**Comment by Ezka77** [comment]
Yeah well if it's a call to a system `mount.nfs`, like it is for `lvm`/`losetup`, there will be not much choice here.

File: Dockerfile (hunk 1 above)
Comment (MikeDombo) [reply Dockerfile:32]: openrc is required to run the nfs service. Without it there are mounting errors the prevent the mount from happening.

**Comment by Ezka77** [comment]
I close your PR, added the requirements to make NFS mounts ; by default it's disabled as the container start without the `privileged` option and the XO server run as `node` user. It can be enabled in docker-compose file (uncomment 2 lines).

**Comment by MikeDombo** [comment]
Thanks for getting it to work and adding this ability. I just tested for myself and it is working great!

Thanks again.

**Pull request closed without merging**